packages/jaeger-ui/src/actions/jaeger-api.js

```diff
@@ -15,11 +15,48 @@
 import { createAction } from 'redux-actions';
 import JaegerAPI from '../api/jaeger';
 
-export const fetchTrace = createAction(
-  '@JAEGER_API/FETCH_TRACE',
-  id => JaegerAPI.fetchTrace(id),
-  id => ({ id })
-);
+export function mergeAll(id) {
+  return traces => {
+    if (traces.data.length < 2) {
+      return traces;
+    }
+    const ot = { traceID: id, spans: [], processes: {}, warnings: [] };
+
+    const processMap = {};
+
+    traces.data.forEach(trace => {
+      const tracePidMap = {};
+      Object.entries(trace.processes).forEach(([pid, process]) => {
+        const pJson = JSON.stringify(process);
+        let targetPid = processMap[pJson];
+        if (!targetPid) {
+          targetPid = `p${Object.keys(processMap).length + 1}`;
+          processMap[pJson] = targetPid;
+          ot.processes[targetPid] = process;
+        }
+        tracePidMap[pid] = targetPid;
+      });
+      trace.spans.forEach(span => {
+        const ns = span;
```

Comment:
I'm guessing this line is to avoid the eslint error `no-param-reassign`, but this will still result in the original span being updated because this is pass by reference.

Regardless, I'm not sure if there is any impact of changing the process IDs from what was originally on the span to a counter. @yurishkuro can you weigh in?

Reply:
If we do stick with the pid -> counter, I think we can improve the performance compared to stringifying every process in every trace.

Reply:
> I'm not sure if there is any impact of changing the process IDs from what was originally on the span to a counter.

I think it's not a counter, but a `"p?"` string where `?` in this case is a counter.

As far as I see, it's to correlate `span` data with the `process` data from a single trace. At that point in the pipeline there haven't been any computations or mappings done to it yet. So they're treated as random string identifiers.

> improve the performance

Typically there's very little code that you can optimize and improve performance in a system. I doubt I've touched any performance critical code. :)

Reply:
The performance concern here comes from `JSON.stringfy`ing every process on all selected traces. I'm not sure how many processes are typical nor their expected sizes, but stringification can be pretty slow so it would be best to avoid.